src/Microsoft.DotNet.VersionTools/Dependencies/FileRegexUpdater.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.VersionTools.Dependencies
+{
+    public abstract class FileRegexUpdater : IDependencyUpdater
+    {
+        public string Path { get; set; }
+        public Regex Regex { get; set; }
```

Comment:
Well for dir.props MSBuild has a project model as well, but I'm OK keeping this for now.  I have seen folks wrestle with this RegExes as well as see accidental inclusion/exclusion due to bad regexes.  I would hope that we could more more to some strongly typed system that understands the documents and is driven by precise lists of updates.